src/visit/Analyzer.cpp

```diff
@@ -1064,6 +1130,16 @@ namespace cmm
 
         node.setDatatype(varContext->getCType());
 
+        // Check if the variable is a const value that we could inline the value
+        // from (if known, such as an enum). For now, only support for enums...
+        // TODO: Support non-enums here.
+        if ((varContext->getModifiers() & EnumModifier::CONST_VALUE) != 0 &&
+             node.getDatatype().type == EnumCType::ENUM && node.getDatatype().pointers == 0)
+        {
+            auto* litteralNode = new LitteralNode(node.getLocation(), varContext->getOptionalValue()->valueEnum);
```

Comment:
Instead of returning a LitteralNode, I propose we return a new node (EnumNode? though behaves like a LitteralNode) to make things more robust going forward.

Reply:
Added EnumUsageNode to resolve this.